### PR TITLE
Decoupled CMS API

### LINKED ISSUE #2872: Decoupled CMS New Doc
Re: /docs/decoupled-cms/
## Issue Description
Customer is trying to build a "decoupled CMS" in their site. In this specific example, index.html in code-root, but won't work because index.php is delivered first by the web server.

## Suggested Resolution
Need to confirm and highlight if this is actually possible or not, or in the future product pipeline.

So far, we can be guided here: 
https://pantheon.io/docs/platform-considerations/#one-application-per-site
https://pantheon.io/docs/platform-considerations/#general-php-framework-support

For confirmation as per last ticket 89023

### PR DESCRIPTION
**Pull request by eabquina**
Done with the introduction points. 

Need to add more content that highlight benefit use of Pantheon platform for CMS API sites. Consider the tools like Redis, Solr and monitoring tools (New Relic).

Example Wordpress site working in Pantheon (Customer site): https://admin.dashboard.pantheon.io/sites/7d39ba3f-2bb4-4f23-b6da-c1ca7d6f0ef3#live/DomainsHTTPS/list
???

Closes #2872 

## Effect
PR includes the following changes:
- New Doc for CMS API
- Share Basics of building CMS as an API
- Highlight benefits of using the Platform for building CMS APIs

## Remaining Work
- [x] Add comments on what to include about platform benefits
- [x] Recommended tools and use cases
- [x] Review uses cases on what can be allowed in Pantheon

### CODE REVIEW

File: source/_docs/cms-api.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+title: CMS as an API or Content as a Service
+description: Consider options on how to build a CMS as an API in the context of decoupled CMS in Pantheon
+tags: [cms]
+categories: []
+---
+
+
+With the rise of Decoupled (**headless**) CMS architecture, the traditionally “monolithic” CMSs  Drupal and WordPress are have now changed. Content as a Service (**CaaS**) means to have an application that manages your content into feeds or services that other applications or properties can consume. It is now easier to implement separate frontend components or frameworks that communicates with the CMS via an API.
+
+A Common feature is that the CMS application exposes a REST API, which allows reading or submitting data over the web. The most common data format used is [JSON](http://www.json.org/), which is supported out-of-the-box for Drupal and WordPress. These together form your "Web Services".
```
Comment (alexfornuto) [inline source/_docs/cms-api.md]: A common feature of what?

File: source/_docs/cms-api.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+You can also write scripts in [Guzzle](http://docs.guzzlephp.org/en/latest/), which is a great tool (and included in Drupal 8 Core) for testing. You can also use cURL via the command line or PHP.
+
+### Pantheon Global CDN
+
+All sites will benefit from the use of [Global CDN](/global-cdn/), a core platform offering for improved performance and security. It enables an advanced caching mechanism, and have free and automated HTTPS by default.
+
+To learn more, read our [Global CDN documentation](/docs/global-cdn/).
+
+<!-- How does Global CDN cache REST API resources -->
```
Comment (alexfornuto) [inline source/_docs/cms-api.md]: What's the status of this commented section?

File: source/_docs/cms-api.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+## Considerations
+
+Pantheon supports only [one application per site](/docs/platform-considerations/#one-application-per-site) and only for Drupal or WordPress applications.
+
+Database access is intended for both debugging or importing large databases. Exposing database connection for external applications use is not recommended. The REST API components of Drupal or WordPress is ideal instead of building your own application. See more information and read our [database access page](/docs/mysql-access/)
+
+### What is REST?
+
+Web Services make it possible for external applications to interact with our application. This aims to enable a **CRUD** for your site, which is to CREATE, READ, UPDATE or DELETE content in your site but through HTTP Requests only. This is different rather than connecting directly to the database.
```
Comment (alexfornuto) [inline source/_docs/cms-api.md]: > This aims to enable a **CRUD** for your site

This _what_?

**Comment by eabquina** [comment]
@alexfornuto I agree and was hoping to get other SMEs to help for these sections. 

For those placeholders, I'm thinking of getting feedback from the others. Or if not, just point to the existing docs for those topics (Build Tools, Global CDN, Solr, Redis Etc)

File: source/_docs/cms-api.md (hunk 2 above)
Comment (eabquina) [reply source/_docs/cms-api.md]: I don't have complete info yet, but would like to get help from other folks for one of these sections

File: source/_docs/cms-api.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+* [Restlet Client](https://chrome.google.com/webstore/detail/restlet-client-rest-api-t/aejoelaoggembcahagimdiliamlcdmfm)
+
+Note: These are Chrome Plugin tools, but there are similar tools for other browsers
+
+### Can I place a single page app in my root directory?
+
+Yes, but instead of using `index.html`, we use `override.html` as the default page to be loaded. It is also possible to have a sub-directory that contains singe-page apps with HTML, JS or CSS files. For deployment, these are treated the same way as other static assets in the platform.
+
+
+## See Also
```
Comment (alexfornuto) [inline source/_docs/cms-api.md]: @eabquina the links below don't point to anything currently on the site.

Comment (eabquina) [reply source/_docs/cms-api.md]: @alexfornuto  I just placed the related blog entries from Pantheon, and references I used for making the docs page.

Should they be removed? Or should I format as external links?

**Comment by eabquina** [comment]
@rachelwhitton thanks for the changes. the page looks much much better now. +1 and thanks for the help

**Comment by alexfornuto** [comment]
I requested a small adjustment to an image, but other than that the copy is 🎆 🎉 💯 :shipit:

**Pull request merged**